tests/test_cli_sync.py

```diff
@@ -242,3 +244,116 @@ def test_sync_dry_run_returns_non_zero_exit_code(runner):
     out = runner.invoke(cli, ["--dry-run"])
 
     assert out.exit_code == 1
+
+
+@mock.patch("piptools.scripts.sync.get_pip_version_for_python_executable")
+@mock.patch("piptools.scripts.sync.get_sys_path_for_python_executable")
+@mock.patch("piptools.scripts.sync.get_installed_distributions")
```

Comment:
There are too much of monkeypatches. Ideally, there should be only `@mock.patch("piptools.sync.run")`.

Reply:
refactored